fineract-provider/src/main/java/org/apache/fineract/notification/domain/NotificationMapperRepository.java

```diff
@@ -18,6 +18,25 @@
  */
 package org.apache.fineract.notification.domain;
 
+import java.util.Collection;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.transaction.annotation.Transactional;
 
-public interface NotificationMapperRepository extends JpaRepository<NotificationMapper, Long> {}
+public interface NotificationMapperRepository extends JpaRepository<NotificationMapper, Long> {
+
+    @Transactional
+    @Modifying(flushAutomatically = true)
```

Comment:
Why the flushAutomatically?

Reply:
Since notifications are being fetching every after few seconds, it desirable that next batch of unread notifications fetched are indeed only those that are unread.

Reply:
I don't get it @wkigenyi . How does a flush help here? Due to the DB isolation level, there will be visibility issues anyway so there's no point in flushing right away.

Reply:
Thanks @galovics flushAutomatically is not needed here. It will not be in the next PR.